src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -564,27 +564,31 @@ TransactionManager::get_extents_if_live(
             std::list<CachedExtentRef> &list) mutable
         {
           auto paddr_seg_id = paddr.as_seg_paddr().get_segment_id();
-          return trans_intr::parallel_for_each(
-            std::move(pin_list),
-            [=, this, &list, &t](
-              LBAPinRef &pin) -> Cache::get_extent_iertr::future<>
-          {
-            auto pin_paddr = pin->get_val();
-            auto &pin_seg_paddr = pin_paddr.as_seg_paddr();
-            auto pin_paddr_seg_id = pin_seg_paddr.get_segment_id();
-            auto pin_len = pin->get_length();
-            if (pin_paddr_seg_id != paddr_seg_id) {
-              return seastar::now();
-            }
-            // Only extent split can happen during the lookup
-            ceph_assert(pin_seg_paddr >= paddr &&
-                        pin_seg_paddr.add_offset(pin_len) <= paddr.add_offset(len));
-            return pin_to_extent_by_type(t, std::move(pin), type
-            ).si_then([&list](auto ret) {
-              list.emplace_back(std::move(ret));
-              return seastar::now();
-            });
-          }).si_then([&list] {
+	  return seastar::do_with(
+	    std::move(pin_list),
+	    [=, this, &list, &t](auto &pin_list) {
```

Comment:
The seastar version doesn't need the `do_with` which allows the container to live on the stack: see https://github.com/scylladb/seastar/blob/dd9577d7e03e33da6337542b2493445ce774df1f/include/seastar/core/loop.hh#L541-L543

This is why I have removed the according `do_with` and don't see any issues. Is there anything missing?

However, I think we probably need to switch to the safer `max_concurrent_for_each` which needs `do_with`, what do you think?

Reply:
I just thought of the following scenario:

```
return seastar::parallel_for_each(container, [](auto &item) {
  return something.then([&item] {
    // do something with item, which might cause segmentation fault, as that item might have been destroyed.
  });
});
```

> However, I think we probably need to switch to the safer `max_concurrent_for_each` which needs `do_with`, what do you think?

I think this will do, but that might need more work from the interruptible future side, can we merge this commit as temporary fix and change to use `max_concurrent_for_each` later?

Reply:
I agree with you that, specific to this part of the code, `do_with` is not necessary, will remove it.

Reply:
I don't have strong opinion whether to remove it or not in this PR. Anyway, we'll add it back with `max_concurrent_for_each`.